core/src/main/java/com/netease/arctic/scan/BasicKeyedTableScan.java

```diff
@@ -149,16 +160,40 @@ private CloseableIterable<ArcticFileScanTask> planChangeFiles() {
   private void split() {
     fileScanTasks.forEach((structLike, fileScanTasks1) -> {
       for (NodeFileScanTask task : fileScanTasks1) {
-        if (task.cost() <= splitSize) {
+        if (task.dataTasks().size() < 2) {
           splitTasks.add(task);
           continue;
         }
-        if (task.dataTasks().size() < 2) {
+
+        if (enableSplitTaskByDelete) {
+          long deleteWeight = task.arcticEquityDeletes().stream().mapToLong(s -> s.file().fileSizeInBytes())
+              .map(s -> s + openFileCost)
+              .sum();
+
+          long dataWeight = task.dataTasks().stream().mapToLong(s -> s.file().fileSizeInBytes())
+              .map(s -> s + openFileCost)
+              .sum();
+          double deleteRatio = deleteWeight * 1.0 / dataWeight;
+
+          if (deleteRatio < splitTaskByDeleteRatio) {
+            long targetSize = Math.min(new Double(deleteWeight / splitTaskByDeleteRatio).longValue(), splitSize);
+            CloseableIterable<NodeFileScanTask> tasksIterable =
```

Comment:
Does the `enableSplitTaskByDelete` only affect the generation of `targetSize`, and is there any consideration for reusing the subsequent split code?